nsis/src/nsis/format/NsisScriptHeader.java

```diff
@@ -17,71 +17,80 @@
 	private int archiveSize;
 	private int compressedHeaderSize;
 	private int flags;
+	private static Structure STRUCTURE;
 
 	public NsisScriptHeader(BinaryReader reader) throws IOException {
-		setMagic(reader.readNextByteArray(NsisConstants.NSIS_MAGIC.length));
+		this.magic = reader.readNextByteArray(NsisConstants.NSIS_MAGIC.length);
 		if (!Arrays.equals(NsisConstants.NSIS_MAGIC, getMagic())) {
-			throw new IOException("not a nsis file.");
+			throw new IOException("Not a nsis file.");
```

Comment:
This should not be an IOException. Consider creating your own Exception subclass, e.g. `nsis.format.InvalidFormat` and throw it on all errors like "well, this format is not as expected".

Reply:
Updated with custom exception